### ISSUE TITLE
Inconsistent results from Diff() - intermittent \u00a0 in result

### ISSUE THREAD

**Issue by dansimone** [google/go-cmp#235]
I've found that the raw, "Go-syntax" results of a Diff() intermittently contain the \u00a0 unicode character rather than a space.

Here's a sample that reproduces this:
```
	type A struct {
		B string
	}
	result := cmp.Diff(A{B: "foo1"}, A{B: "foo2"})
        // Print out the Go-syntax output:
	fmt.Println("Result is: " + fmt.Sprintf("%#v", result))
```

Running the above sometimes produces:
```Result is: "\u00a0\u00a0reconciler.A{\n-\u00a0\tB: \"foo1\",\n+\u00a0\tB: \"foo2\",\n\u00a0\u00a0}\n"```

and sometimes produces: ```Result is: "  reconciler.A{\n- \tB: \"foo1\",\n+ \tB: \"foo2\",\n  }\n"``

Result is: "\u00a0\u00a0reconciler.A{\n-\u00a0\tB: \"foo1\",\n+\u00a0\tB: \"foo2\",\n\u00a0\u00a0}\n"

This problem is only apparent if you are looking at the _Go-syntax_ (%#v) output, and is not apparent by just printing the string.  However, there are libraries that look at the Go-syntax value of a string - for example, the various string comparison methods of https://github.com/stretchr/testify.  Attempting to use the output of Diff() in test cases with https://github.com/stretchr/testify is not usable due to this inconsistency.

**Comment by neild** [comment]
This is by design. As the documentation for `Diff` states:

> Do not depend on this output being stable. If you need the ability to programmatically interpret the difference, consider using a custom Reporter.

**Comment by dsnet** [comment]
Closing as working as intended.

The first few versions of `cmp` unfortunately kept breaking a number of tests that assumed that the output would never change every time the reporter output was changed. At some point we made it random precisely so that brittle tasks cannot be written. The fact that the reporter output has been improved significantly in recent versions is thanks to the freedom to make beneficial changes made possible by deliberate randomization.
